src/lib/linux-api/src/signal.rs

```diff
@@ -960,6 +994,58 @@ impl sigaction {
     const SIG_DFL: usize = 0;
     const SIG_IGN: usize = 1;
 
+    /// Consider [`sigaction::new_with_default_restorer`] instead, which takes
+    /// care of setting up a restorer.
+    pub fn new_raw(
+        handler: SignalHandler,
+        flags: SigActionFlags,
+        mask: sigset_t,
+        restorer: Option<unsafe extern "C" fn()>,
+    ) -> Self {
+        let handler = match handler {
+            SignalHandler::Handler(h) => Some(h),
+            SignalHandler::Action(a) => {
+                assert!(flags.contains(SigActionFlags::SA_SIGINFO));
+                Some(unsafe {
+                    core::mem::transmute::<
+                        unsafe extern "C" fn(i32, *mut siginfo_t, *mut core::ffi::c_void),
+                        unsafe extern "C" fn(i32),
+                    >(a)
```

Comment:
Can this use the `SignalHandlerFn` and `SignalActionFn` like in the transmute below? And maybe a comment explaining why this is safe?

Reply:
done